[bug] Save and restore packages from the cache failing when saving in windows and restoring in unix

### Environment details

Save side:
* Operating System+version: Windows 10
* Compiler+version: Visual Studio 16 2019 (MSVC 19.2)
* Conan version: 2.0.14
* Python version: 3.10.11

Restore side:
* Operating System+version: macos ventura 13.6.1 (22G313)
* Compiler+version: Apple clang version 15.0.0
* Conan version: 2.0.14
* Python version: 3.11.6

### Steps to reproduce

1. Create a package in windows. I've tried to create different packages with different libraries
2. Try to export using the conan cache restore command in a windows environment

Exporting the same package in linux or antoher mac device is not failing. 
I don't know if doing this process the opposite way (save in windows and export in unix does fail)

### Logs

-> % conan cache restore windows-x86_64.tgz
				ERROR: Traceback (most recent call last):
				  File "/usr/local/Cellar/conan/2.0.14/libexec/lib/python3.12/site-packages/conan/cli/cli.py", line 272, in main
				    cli.run(args)
				  File "/usr/local/Cellar/conan/2.0.14/libexec/lib/python3.12/site-packages/conan/cli/cli.py", line 172, in run
				    command.run(self._conan_api, args[0][1:])
				  File "/usr/local/Cellar/conan/2.0.14/libexec/lib/python3.12/site-packages/conan/cli/command.py", line 141, in run
				    sub.run(conan_api, parser, *args)
				  File "/usr/local/Cellar/conan/2.0.14/libexec/lib/python3.12/site-packages/conan/cli/command.py", line 155, in run
				    info = self._method(conan_api, parent_parser, self._parser, *args)
				           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
				  File "/usr/local/Cellar/conan/2.0.14/libexec/lib/python3.12/site-packages/conan/cli/commands/cache.py", line 146, in cache_restore
				    package_list = conan_api.cache.restore(path)
				                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
				  File "/usr/local/Cellar/conan/2.0.14/libexec/lib/python3.12/site-packages/conan/api/subapi/cache.py", line 162, in restore
				    assert rel_path == recipe_folder, f"{rel_path}!={recipe_folder}"
				AssertionError: p/utilicc39abe1a64b1!=p\utilicc39abe1a64b1
 
				ERROR: p/utilicc39abe1a64b1!=p\utilicc39abe1a64b1

--- a/conan/api/subapi/cache.py
+++ b/conan/api/subapi/cache.py
@@ -119,6 +119,7 @@ class CacheAPI:
             for ref, ref_bundle in package_list.refs().items():
                 ref_layout = app.cache.recipe_layout(ref)
                 recipe_folder = os.path.relpath(ref_layout.base_folder, cache_folder)
+                recipe_folder = recipe_folder.replace("\\", "/")  # make win paths portable
                 ref_bundle["recipe_folder"] = recipe_folder
                 out.info(f"Saving {ref}: {recipe_folder}")
                 tgz.add(os.path.join(cache_folder, recipe_folder), recipe_folder, recursive=True)
@@ -126,11 +127,13 @@ class CacheAPI:
                     pref_layout = app.cache.pkg_layout(pref)
                     pkg_folder = pref_layout.package()
                     folder = os.path.relpath(pkg_folder, cache_folder)
+                    folder = folder.replace("\\", "/")  # make win paths portable
                     pref_bundle["package_folder"] = folder
                     out.info(f"Saving {pref}: {folder}")
                     tgz.add(os.path.join(cache_folder, folder), folder, recursive=True)
                     if os.path.exists(pref_layout.metadata()):
                         metadata_folder = os.path.relpath(pref_layout.metadata(), cache_folder)
+                        metadata_folder = metadata_folder.replace("\\", "/")  # make paths portable
                         pref_bundle["metadata_folder"] = metadata_folder
                         out.info(f"Saving {pref} metadata: {folder}")
                         tgz.add(os.path.join(cache_folder, metadata_folder), metadata_folder,
@@ -159,6 +162,7 @@ class CacheAPI:
             recipe_layout = cache.get_or_create_ref_layout(ref)
             recipe_folder = ref_bundle["recipe_folder"]
             rel_path = os.path.relpath(recipe_layout.base_folder, cache.cache_folder)
+            rel_path = rel_path.replace("\\", "/")
             assert rel_path == recipe_folder, f"{rel_path}!={recipe_folder}"
             out.info(f"Restore: {ref} in {recipe_folder}")
             for pref, pref_bundle in package_list.prefs(ref, ref_bundle).items():

--- a/conan/api/subapi/cache.py
+++ b/conan/api/subapi/cache.py
@@ -135,7 +135,7 @@ class CacheAPI:
                         metadata_folder = os.path.relpath(pref_layout.metadata(), cache_folder)
                         metadata_folder = metadata_folder.replace("\\", "/")  # make paths portable
                         pref_bundle["metadata_folder"] = metadata_folder
-                        out.info(f"Saving {pref} metadata: {folder}")
+                        out.info(f"Saving {pref} metadata: {metadata_folder}")
                         tgz.add(os.path.join(cache_folder, metadata_folder), metadata_folder,
                                 recursive=True)
             serialized = json.dumps(package_list.serialize(), indent=2)

--- a/conans/test/integration/command_v2/test_cache_save_restore.py
+++ b/conans/test/integration/command_v2/test_cache_save_restore.py
@@ -1,6 +1,7 @@
 import json
 import os
 import shutil
+import tarfile
 
 from conans.test.assets.genconanfile import GenConanfile
 from conans.test.utils.tools import TestClient
@@ -17,6 +18,16 @@ def test_cache_save_restore():
     cache_path = os.path.join(c.current_folder, "conan_cache_save.tgz")
     assert os.path.exists(cache_path)
     _validate_restore(cache_path)
+
+    # Lets test that the pkglist does not contain windows backslash paths to make it portable
+    with open(cache_path, mode='rb') as file_handler:
+        the_tar = tarfile.open(fileobj=file_handler)
+        fileobj = the_tar.extractfile("pkglist.json")
+        pkglist = fileobj.read()
+        the_tar.close()
+
+    package_list = json.loads(pkglist)
+    assert "\\" not in package_list
 
 
 def test_cache_save_downloaded_restore():

